src/renderer/shared/api/xcm/xcmService.ts

```diff
@@ -316,27 +323,46 @@ export const getVersionedAccountLocation = (
 };
 
 export const getAccountLocation = (accountId?: AccountId): Object | undefined => {
-  return {
-    parents: 0,
-    interior: {
-      X1: {
+  const isEthereum = isEthereumAccountId(accountId);
+
+  const account = isEthereum
+    ? {
+        accountKey20: {
+          network: 'Any',
+          key: accountId,
+        },
+      }
+    : {
         accountId32: {
           network: 'Any',
           id: accountId,
         },
-      },
+      };
+
+  return {
+    parents: 0,
+    interior: {
+      X1: account,
     },
   };
 };
 
 const getChildLocation = (parachainId: number, accountId?: AccountId): Object => {
   const location: Record<string, any> = { parachainId };
+  const isEthereum = isEthereumAccountId(accountId);
 
   if (accountId) {
-    location.accountId = {
-      network: 'Any',
-      id: accountId,
-    };
+    if (isEthereum) {
+      location.accountKey = {
```

Comment:
You could inline condition with property destructuring 
`...(isEth ? {key: accountId} : {id: accountId})`
Or this one
`[isEth ? 'key' : 'id']: accountId`

Reply:
I can't do this, because we have not only `key` and `id` differance, but `accountKey` and `accountId` on higher level